core/trino-main/src/main/java/io/trino/connector/system/TableCommentSystemTable.java

```diff
@@ -122,19 +124,19 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
                         prefix.getSchemaName(),
                         relationNames -> accessControl.filterTables(session.toSecurityContext(), catalog, relationNames));
 
+                Map<SchemaTableName, TableMetadata> tables = getRedirectionAwareTables(session, prefix);
```

Comment:
Is this listing all tables matching prefix the second time?

Reply:
indeed `io.trino.plugin.iceberg.catalog.glue.TestIcebergGlueCatalogAccessOperations#testSystemMetadataTableComments` regresses. 

not only listing tables is redundant; getRedirectionAwareTables acquires a table handle for every each of them, and getting table handle is costly. This incurs additional GetTable call (avoidable when number of table is small and we can cache all them in per-query cache... which we should assume to be unbounded) and additional S3 trips.